extensions/azure-openai-oauth-pyshiny/app.py

```diff
@@ -0,0 +1,105 @@
+import os
+from shiny import App, ui, render, Inputs, Outputs
+from shiny.session._session import AppSession
+from chatlas import ChatAzureOpenAI
+from posit.connect import Client
+from posit.connect.errors import ClientError
+from posit.connect.oauth.oauth import OAuthTokenType
+
+setup_ui = ui.page_fillable(
+    ui.div(
+        ui.div(
+            ui.card(
+                ui.card_header(
+                    ui.h2("Setup Required")
+                    ),
+                ui.card_body(
+                    ui.p(
+                        ui.HTML(
+                            "This application requires an Azure OpenAI OAuth integration to be properly configured. "
+                            "For more detailed instructions, please refer to the "
+                            '<a href="https://docs.posit.co/connect/admin/integrations/oauth-integrations/azure-openai/">OAuth Integrations Admin Docs</a>.'
+                        )
+                    )
+                )
+            ),
+            style="max-width: 600px; margin: 0 auto; position: absolute; top: 50%; left: 50%; transform: translate(-50%, -50%);"
+        )
+    ),
+    fillable = True
+)
+
+app_ui = ui.page_fillable(
+    ui.div(
+        ui.div(
+            ui.card(
+                ui.card_header(
+                    ui.h3("Palmer Penguins Chat Assistant")
+                ),
+                ui.card_body(
+                    ui.chat_ui("chat", placeholder = "Enter a message...", height = "300px")
+                )
+            ),
+            style="max-width: 800px; margin: 0 auto; position: absolute; top: 50%; left: 50%; transform: translate(-50%, -50%); width: 80%;"
+        )
+    )
+)
+
+screen_ui = ui.page_output("screen")
+
+connect_server = os.getenv("CONNECT_SERVER")
+
+def server(input: Inputs, output: Outputs, app_session: AppSession):
+
+    user_session_token = app_session.http_conn.headers.get(
+        "Posit-Connect-User-Session-Token"
+    )
+
+    OAUTH_INTEGRATION_ENABLED = True
+    if user_session_token:
+        try:
+            client = Client().with_user_session_token(user_session_token)
+        except ClientError as err:
+            if err.error_code == 212:
+                OAUTH_INTEGRATION_ENABLED = False
+    
+    if OAUTH_INTEGRATION_ENABLED:
+        client = Client(url = connect_server)
+
+        credentials = client.oauth.get_credentials(
+            user_session_token=user_session_token,
+            requested_token_type=OAuthTokenType.ACCESS_TOKEN  # Optional
+        )
+
+        chat = ChatAzureOpenAI(
+            endpoint="https://8ul4l3wq0g.openai.azure.com",
+            deployment_id="gpt-4o-mini",
+            api_version="2024-12-01-preview",
```

Comment:
I had a similar question on #202. This looks like data specific to the Posit organization, and not something to be used by others. Can this app be added to the Gallery generally for others to use? The Gallery doesn't let others change the content prior to publishing.